lib/ProductOpener/DataQualityFood.pm

```diff
@@ -758,6 +761,37 @@ sub check_nutrition_data ($product_ref) {
 			if (($nid eq 'fat') or ($nid eq 'carbohydrates') or ($nid eq 'proteins') or ($nid eq 'salt')) {
 				$total += $product_ref->{nutriments}{$nid . "_100g"};
 			}
+
+			# variables to check if there are 3 or more duplicates in nutriments
+			if (
+				(
+					   ($nid eq 'fat_100g')
+					or ($nid eq 'saturated-fat_100g')
+					or ($nid eq 'carbohydrates_100g')
+					or ($nid eq 'sugars_100g')
+					or ($nid eq 'fiber_100g')
+					or ($nid eq 'proteins_100g')
+					or ($nid eq 'salt_100g')
+					or ($nid eq 'sodium_100g')
+				)
+				and ($product_ref->{nutriments}{$nid} > 1)
+				)
+			{
+				push(@major_nutriments_values, $product_ref->{nutriments}{$nid});
+			}
+
+		}
+
+		# create a hash key: nutriment value, value: number of occurence
+		foreach my $nutriment_value (@major_nutriments_values) {
+			$nutriments_values_occurences{$nutriment_value}++;
```

Comment:
This works but Perl will emit a warning that we are incrementing an undefined value the first time, so it's better to check if it exists and set it to 0 otherwise first.